### PR TITLE
June 2020 Update

### PR DESCRIPTION
**Pull request by sesham18**
Please review the following MCW workshop addition: HOL-step-by-step - oracle-to-postgresql.md.

### CODE REVIEW

**Comment by DawnmarieDesJardins** [comment]
Based on the new exercised being added to this workshop, Solliance recommends it be renamed at the completion of the refresh.  The suggested new title is:  **Migrating Oracle to Azure databases**

Please let us know what you think.

**Comment by DawnmarieDesJardins** [comment]
@timahenning 
Spoke with Nabeel regarding an updated title on this one.  Since we have two possible paths in the lab, his suggestion is:
Migrating Oracle to Azure SQL or PostgreSQL

Requesting your opinion on the change.
Dawnmarie

File: Hands-on lab/Before the HOL - Migrating Oracle to Azure SQL and PostgreSQL.md [hunk 1]
```diff
@@ -0,0 +1,527 @@
...
+
+    ![Resource groups is selected in the Azure navigation pane, "hands" is entered into the filter box, and the "hands-on-lab-SUFFIX" resource group is highlighted.](./media/resource-groups.png "Resource groups list")
+
+3. In the list of resources for your resource group, select the LabVM Virtual Machine.
+
+    ![The list of resources in the hands-on-lab-SUFFIX resource group are displayed, and LabVM is highlighted.](./media/resource-group-resources-labvm.png "LabVM in resource group list")
+
+4. On your Lab VM blade, select **Connect** from the top menu.
+
+    ![The LabVM blade is displayed, with the Connect button highlighted in the top menu.](./media/connect-labvm.png "Connect to LabVM")
```
Comment (DawnmarieDesJardins) [inline Hands-on lab/Before the HOL - Migrating Oracle to Azure SQL and PostgreSQL.md:301]: Please check image on this line.  It's not resolving when I view the document.  Thx!

Comment (timahenning) [reply Hands-on lab/Before the HOL - Migrating Oracle to Azure SQL and PostgreSQL.md]: Fixed

**Comment by DawnmarieDesJardins** [comment]
Migrating Oracle to Azure Lab (QC question)
Exercise 4, task 1, number 1, line 766, I'm I correct in think this link needs to be changed with the title change?  Please review.

1. Connect to your Lab VM, as you did in Task 5 of the [Before the Hands-on Lab](./Before%20the%20HOL%20-%20Data%20Platform%20upgrade%20and%20migration.md#task-5-connect-to-the-lab-vm) exercise.

File: Hands-on lab/HOL step-by-step - Migrating Oracle to PostgreSQL.md [hunk 2]
```diff
@@ -0,0 +1,1439 @@
...
+
+7. Select **Finish** when the installation is complete.
+
+### Task 4: Create the Northwind database in Oracle 11g XE
+
+WWI has provided you with a copy of their application, including a database script to create their Oracle database. They have asked that you use this as a starting point for migrating their database and application to Azure Database for PostgreSQL. In this task, you will create a connection to the Oracle database on your Lab VM, and create a database called Northwind.
+
+1. Navigate to the Lab VM and download the starter project by downloading a .zip copy of the Data Platform upgrade and migration project from the GitHub repo.
+
+2. . In a web browser, navigate to the [Data Platform upgrade and migration MCW repo](https://github.com/Microsoft/MCW-Data-Platform-upgrade-and-migration)
```
Comment (DawnmarieDesJardins) [inline Hands-on lab/HOL step-by-step - Migrating Oracle to PostgreSQL.md:219]: Remove the extra period after 2. . 
Does this link need to be changed with the new title?
Period at the end of the sentence is missing

File: Hands-on lab/HOL step-by-step - Migrating Oracle to PostgreSQL.md [hunk 3]
```diff
@@ -0,0 +1,1439 @@
...
+
+11. Select **Test Connection** to verify the settings are correct, and select **OK** to close the popup.
+
+12. Select **OK** to create the Database Connection.
+
+13. You will now see the Northwind connection in the Database Explorer window.
+
+    ![The Northwind connection is selected in the Database Explorer window.](./media/visual-studio-fusion-database-explorer.png "View the Northwind connection")
+
+14. In Visual Studio, select **File** in the menu, then select **Open File**, and navigate to `C:\handsonlab\MCW-Data-Platform-upgrade-and-migration-master\Hands-on lab\lab-files\starter-project\Oracle Scripts\`, select the file `1.northwind.oracle.schema.sql`, and then select **Open**.
```
Comment (DawnmarieDesJardins) [inline Hands-on lab/HOL step-by-step - Migrating Oracle to PostgreSQL.md:262]: Does this link need to be changed with re-title?

File: Hands-on lab/HOL step-by-step - Migrating Oracle to PostgreSQL.md [hunk 4]
```diff
@@ -0,0 +1,1439 @@
...
+
+    ![The information above is entered and highlighted in the Database Connection Properties * Oracle dialog box, and Test Connection is selected at the bottom.](./media/visual-studio-database-explorer-modify-connection-update.png "Specify the settings")
+
+20. Select **OK** to close the Database Connection Properties dialog.
+
+21. Select the **Open File** icon on the Visual Studio toolbar.
+
+    ![The Open File icon is highlighted on the Visual Studio toolbar.](./media/visual-studio-toolbar-open-file.png "Select Open File")
+
+22. In the Open File dialog, navigate to `C:\handsonlab\MCW-Data-Platform-upgrade-and-migration-master\Hands-on lab\lab-files\starter-project\Oracle Scripts`, select the file `2.northwind.oracle.tables.views.sql`, and then select **Open**.
```
Comment (DawnmarieDesJardins) [inline Hands-on lab/HOL step-by-step - Migrating Oracle to PostgreSQL.md:295]: Does this link need to be changed with re-title?

File: Hands-on lab/HOL step-by-step - Migrating Oracle to PostgreSQL.md [hunk 5]
```diff
@@ -0,0 +1,1439 @@
...
+
+## Exercise 2: Assess the Oracle 11g Database before Migrating to PostgreSQL
+
+Duration: 15 mins
+
+In this exercise, you will prepare the existing Oracle database for its migration to PostgreSQL. Preparation involves two main steps. The first step is to update the database statistics. Statistics about the database become outdated as data volumes and activity change over time. Second, you will need to identify invalid objects in the Oracle database. The migration utility will not migrate invalid objects.
+
+### Task 1: Update Statistics and Identify Invalid Objects
+
+1. Create a new folder titled `Postgre Scripts` at the `C:\handsonlab\MCW-Data-Platform-upgrade-and-migration\Hands-on lab\lab-files\starter-project` location.
```
Comment (DawnmarieDesJardins) [inline Hands-on lab/HOL step-by-step - Migrating Oracle to PostgreSQL.md:335]: Does this path need to be changed with re-title?

File: Hands-on lab/HOL step-by-step - Migrating Oracle to PostgreSQL.md [hunk 6]
```diff
@@ -0,0 +1,1439 @@
...
+3. Now, you will populate the new file with the following statements:
+
+    ```sql
+    -- 11g script
+    EXECUTE DBMS_STATS.GATHER_SCHEMA_STATS(ownname => 'NW');
+    EXECUTE DBMS_STATS.GATHER_DATABASE_STATS;
+    EXECUTE DBMS_STATS.GATHER_DICTIONARY_STATS;
+    ```
+
+4. Save the file as `update-llg-stats.sql` in the `C:\handsonlab\MCW-Data-Platform-upgrade-and-migration\Hands-on lab\lab-files\starter-project\Postgre Scripts` directory. Run the file as you did when you created database objects.
```
Comment (DawnmarieDesJardins) [inline Hands-on lab/HOL step-by-step - Migrating Oracle to PostgreSQL.md:350]: Does this path need to be changed with re-title?

File: Hands-on lab/HOL step-by-step - Migrating Oracle to PostgreSQL.md [hunk 7]
```diff
@@ -0,0 +1,1439 @@
...
+
+5. At the **New** page, navigate to **Web** under **Azure Marketplace**. Select **Web App**.
+
+    ![Navigating to the Web App option on Azure Marketplace.](./media/creating-web-app.png "Web app option highlighted on Marketplace")
+
+6. Create a new app in your hands-on-lab-SUFFIX resource group. Configure it with the following parameters. Keep all other settings at their default values. Select **Review + create** when you are ready.
+
+    - **Name**: Configure a unique app name (the name you choose will form part of your app's URL).
+    - **Runtime stack**: ASP.NET V4.7.
+    - **Region**: Must support all necessary resources.
```
Comment (DawnmarieDesJardins) [inline Hands-on lab/HOL step-by-step - Migrating Oracle to PostgreSQL.md:406]: This is a dropdown.  Will this make sense to everyone or should it say something like:
Choose a region closest to you the supports all necessary resources.

File: Hands-on lab/HOL step-by-step - Migrating Oracle to PostgreSQL.md [hunk 8]
```diff
@@ -0,0 +1,1439 @@
...
+
+    ![Configuring the web app details.](./media/web-app-configuration.PNG "Project Details window with pertinent details")
+
+7. Select **Create** after reviewing parameters. Once the deployment finishes, navigate to the **App Service** resource you created. Select **Get publish profile** under the resource's **Overview** page.
+
+    ![Downloading the publish profile.](./media/get-app-publish-file.png "Get publish profile")
+
+8. Save the file and move it to `C:\handsonlab\MCW-Data-Platform-upgrade-and-migration\Hands-on lab\lab-files\starter-project`. Later, we will need this file to import into Visual Studio for deployment.
```
Comment (DawnmarieDesJardins) [inline Hands-on lab/HOL step-by-step - Migrating Oracle to PostgreSQL.md:414]: Does this path need to be updated?

File: Hands-on lab/HOL step-by-step - Migrating Oracle to PostgreSQL.md [hunk 9]
```diff
@@ -0,0 +1,1439 @@
...
+
+5. pgAdmin will prompt you to set a password to govern access to database credentials. Enter a password. Confirm your choice. For now, our configuration of pgAdmin is complete.
+
+### Task 4: Install ora2pg
+
+**Ora2pg** is the tool we will use to migrate database objects and data. Microsoft's Data Migration Team has greatly simplified the process of obtaining this tool by providing the **installora2pg.ps1** script. You can download using the link below:
+
+ **Download**: <https://github.com/microsoft/DataMigrationTeam/blob/master/IP%20and%20Scripts/PostgreSQL%20Migration%20and%20Assessment%20Tools/installora2pg.ps1>.
+
+1. Copy Microsoft's script to the `C:\handsonlab\MCW-Data-Platform-upgrade-and-migration\Hands-on lab\lab-files\starter-project\Postgre Scripts` location.
```
Comment (DawnmarieDesJardins) [inline Hands-on lab/HOL step-by-step - Migrating Oracle to PostgreSQL.md:468]: Does this path need to be changed?

File: Hands-on lab/HOL step-by-step - Migrating Oracle to PostgreSQL.md [hunk 10]
```diff
@@ -0,0 +1,1439 @@
...
+
+38. Congratulations! You have successfully migrated data into the new database. If you return to the **NW** schema page, and select the **Full load** tab, you will see a count of the number of rows exported from a table in the Oracle database to its equivalent in the PostgreSQL database. This is our result for the first ten migrated tables.
+
+    ![Screenshot verifying that the correct number of tables were exported.](./media/migrated-row-counts.png "Number of rows copied from source to target")
+
+## Task 3: Finishing the table schema migration
+
+We migrated the data before the constraints to reduce the time required to copy data into the tables. In addition, if foreign keys were present on the target tables, data migration would fail. So, in this task, we will add constraints, foreign keys, and indexes to the target tables.
+
+1. Then, layer on constraints (not foreign keys).
```
Comment (DawnmarieDesJardins) [inline Hands-on lab/HOL step-by-step - Migrating Oracle to PostgreSQL.md:920]: This is the first instruction for Task 3 but the sentence starts with Then.
Is that correct?

File: Hands-on lab/HOL step-by-step - Migrating Oracle to PostgreSQL.md (hunk 7 above)
Comment (timahenning) [reply Hands-on lab/HOL step-by-step - Migrating Oracle to PostgreSQL.md:406]: Some regions do not support this lab.  That is why lab trainers and participants need to test.  We cannot say if the closest region will work.

File: Hands-on lab/HOL step-by-step - Migrating Oracle to PostgreSQL.md [hunk 11]
```diff
@@ -0,0 +1,1439 @@
...
+    FROM employees e
+     JOIN employeeterritories et ON e.employeeid = et.employeeid
+     JOIN territories t ON et.territoryid = t.territoryid;
+    ```
+
+6. Now, execute the query by selecting the **execution** button on the toolbar.
+
+    ![Screenshot showing running of the query.](./media/running-query.png "Execution button")
+
+7. If you were successful, you should see an output similar to the following. Notice the following result set, which consists of 49 rows. It is available under the **Data Output** tab.
```
Comment (DawnmarieDesJardins) [inline Hands-on lab/HOL step-by-step - Migrating Oracle to PostgreSQL.md:956]: Notice the following result set, which consists of 49 rows.
Sorry, this is confusing to me...notice what?  "Review the following..." maybe?

File: Hands-on lab/HOL step-by-step - Migrating Oracle to PostgreSQL.md (hunk 2 above)
Comment (timahenning) [reply Hands-on lab/HOL step-by-step - Migrating Oracle to PostgreSQL.md:219]: Once the new repo is live, I can quickly change the URL's.

File: Hands-on lab/HOL step-by-step - Migrating Oracle to PostgreSQL.md [hunk 12]
```diff
@@ -0,0 +1,1439 @@
...
+In this task, we will be recreating the ADO.NET data models to accurately represent our PostgreSQL database objects. Entity Framework leverages ADO.NET, allowing us to map database objects to classes.  
+
+1. First, install the Entity Framework.
+
+    - Navigate to the Package Manager console and enter the following command:
+
+    ```cmd
+    Install-Package EntityFramework
+    ```
+
```
Comment (DawnmarieDesJardins) [inline Hands-on lab/HOL step-by-step - Migrating Oracle to PostgreSQL.md:1089]: Are we missing an instruction?  1. says to enter a command.  Underneath the command is an image that shows Tools, NuGet Package Manager & Package Manager Console instructions.  It doesn't match the instruction I don't think....

File: Hands-on lab/HOL step-by-step - Migrating Oracle to PostgreSQL.md [hunk 13]
```diff
@@ -0,0 +1,1457 @@
...
+© 2020 Microsoft Corporation. All rights reserved.
+
+Microsoft and the trademarks listed at <https://www.microsoft.com/en-us/legal/intellectualproperty/Trademarks/Usage/General.aspx> are trademarks of the Microsoft group of companies. All other trademarks are property of their respective owners.
+
+**Contents**
+
+- [Migrating Oracle to PostgreSQL hands-on lab step-by-step](#migrating-oracle-to-postgresql-hands-on-lab-step-by-step)
```
Comment (DawnmarieDesJardins) [inline Hands-on lab/HOL step-by-step - Migrating Oracle to PostgreSQL.md:27]: I broke this hotlink when I took out the Azure SQL.  I can't figure out what's wrong.  Please fix. (Sorry)

File: Whiteboard design session/WDS trainer guide - Migrating Oracle to Azure SQL.md [hunk 14]
```diff
@@ -41,7 +41,7 @@ Microsoft and the trademarks listed at <https://www.microsoft.com/en-us/legal/in
   - [Step 3: Present the solution](#step-3-present-the-solution)
   - [Wrap-up](#wrap-up)
   - [Additional references](#additional-references)
-- [Data Platform upgrade and migration whiteboard design session trainer guide](#data-platform-upgrade-and-migration-whiteboard-design-session-trainer-guide-1)
+- [Data Platform upgrade and migration whiteboard design session trainer guide](#data-platform-upgrade-and-migration-whiteboard-design-session-trainer-guide)
```
Comment (DawnmarieDesJardins) [inline Whiteboard design session/WDS trainer guide - Migrating Oracle to Azure SQL.md:44]: Rename to Migrating Oracle to Azure SQL please.  (I would but I keep breaking the darn links when I do it!)

File: Whiteboard design session/WDS trainer guide - Migrating Oracle to PostgreSQL.md [hunk 15]
```diff
@@ -0,0 +1,625 @@
...
+
+**Contents**
+
+- [Migrating Oracle to PostgreSQL whiteboard design session trainer guide](#migratingoracletopostgresql-whiteboard-design-session-trainer-guide)
+- [Trainer information](#trainer-information)
+  - [Role of the trainer](#role-of-the-trainer)
+  - [Whiteboard design session flow](#whiteboard-design-session-flow)
+  - [Before the whiteboard design session: How to prepare](#before-the-whiteboard-design-session-how-to-prepare)
+  - [During the whiteboard design session: Tips for an effective whiteboard design session](#during-the-whiteboard-design-session-tips-for-an-effective-whiteboard-design-session)
+- [Migrating Oracle to PostgreSQL whiteboard design session student guide](#migratingoracletopostgresql-whiteboard-design-session-student-guide)
```
Comment (DawnmarieDesJardins) [inline Whiteboard design session/WDS trainer guide - Migrating Oracle to PostgreSQL.md:34]: Hotlink broken, please review.

File: Hands-on lab/HOL step-by-step - Migrating Oracle to PostgreSQL.md (hunk 12 above)
Comment (timahenning) [reply Hands-on lab/HOL step-by-step - Migrating Oracle to PostgreSQL.md:1104]: No, the instructions are correct.  I switch the sub step around to make it a little clearer.

**Pull request merged**